### PR TITLE
[Android] fixes lint errors, removed unused strings and files

### PR DESCRIPTION
**Pull request by alexsafe**
Resolves 

## Submitter Checklist:

- [ x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -178,9 +178,6 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
       <message name="IDS_ACCESSIBILITY_TOOLBAR_BTN_BRAVE_REWARDS" desc="Content description for the button that is enable, disable Brave Rewards.">
         Manage Brave Rewards
       </message>
-      <message name="IDS_PREFS_SECTION_SEARCH_ENGINE" desc="Title for search engine section that can select DSE of normal and private tab">
```
Comment (samartnik) [inline browser/ui/android/strings/android_brave_strings.grd:181]: Are all these strings not used?

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -257,7 +233,7 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
         Block trackers &amp; ads
       </message>
       <message name="IDS_BLOCK_TRACKERS_ADS_SUMMARY" desc="Text summary for Block trackers and ads.">
-        Prevents ads, popups, and trackers from loading
```
Comment (samartnik) [inline browser/ui/android/strings/android_brave_strings.grd:260]: Not sure what warning is fixed by this change.

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (alexsafe) [reply browser/ui/android/strings/android_brave_strings.grd:181]: yes, not used anymore (some were leftovers from a previous design too). i went through the warnings to identify all of them

File: browser/ui/android/strings/android_brave_strings.grd (hunk 2 above)
Comment (alexsafe) [reply browser/ui/android/strings/android_brave_strings.grd:260]: Not a warning fix, i realised the text wasn't complete as per the design specs

**Review by samartnik — APPROVED** [review]
lgtm

**Pull request merged**